sos/policies/__init__.py

```diff
@@ -922,5 +948,239 @@ def pre_work(self):
 
         return
 
+    def prompt_for_upload_user(self):
+        """Should be overridden by policies to determine if a user needs to
+        be provided or not
+        """
+        if (not self.commons['cmdlineopts'].upload_user and not
+                self._upload_user):
+            msg = "Please provide upload user for %s: " % self.get_upload_url()
+            self.upload_user = input(msg)
+
+    def prompt_for_upload_password(self):
+        """Should be overridden by policies to determine if a password needs to
+        be provided for upload or not
+        """
+        if (not self.upload_password and not
+                self._upload_password) and self.upload_user:
+            msg = (
+                "Please provide the upload password for %s: "
+                % self.upload_user or self._upload_user
```

Comment:
Due to the test 3 lines above, `self.upload_user` is evaluated to `True`, hence the `.. or self._upload_user` is redundant.

Reply:
Ack, fixed.